machine/machine_core/machine.py

```diff
@@ -358,8 +358,11 @@ def wait_for_cockpit_running(self, address="localhost", port=9090, seconds=30, t
             self.execute(script=WAIT_COCKPIT_RUNNING)
 
     def curl(self, *args, headers=None):
-        header_args = []
+        cmd = ['curl', '--no-progress-meter']
+        if self.image in ['rhel-8-6', 'centos-8-stream']:
+            cmd.remove('----no-progress-meter')  # not available in old curl
+            cmd.extend(['--silent', '--show-error'])
```

Comment:
Ugh -- what's wrong with just always using `silent --show-error`?

Reply:
I'd agree with this except that I specifically want to make this stick out like a sore thumb so that we can remove it when we drop support for RHEL8.

Reply:
(I'm pecking on this because I don't really like curl to behave differently when testing across different OSes.)

Reply:
Are you making your argument on the basis that `--silent` `--show-error` is exactly equivalent to `--no-progress-meter` or are you making it on the basis that it might not be? :)

Reply:
It's about "I don't know and I don't want to". Either it is exactly equivalent, then we can use the options which work everywhere. Or it's not, then we'd do differnet things on different OSes. So it doesn't matter, I wouldn't want this conditional either way.

Reply:
Okay.  We can do it your way :)